stan/math/prim/mat/prob/neg_binomial_2_log_glm_lpmf.hpp

```diff
@@ -91,6 +87,10 @@ return_type_t<T_x, T_alpha, T_beta, T_precision> neg_binomial_2_log_glm_lpmf(
     check_consistent_sizes(function, "Vector of intercepts", alpha,
                            "Vector of dependent variables", y);
   }
+  check_nonnegative(function, "Failures variables", y);
```

Comment:
What is the reasoning behind these changes? Not saying it shouldnt be done, just a clarification.

Reply:
The same as with changes I made to other GLMs when writing GPU implementations. I think mismatched sizes is a larger error than wrong values in matrices so they should be checked and reported first.

Reply:
OK, once again I agree. Please write that down in the PR text instead of " There are also some minor changes". I know its minor but as you can see I get confused every time :)